src/Foundation/NSDate.cs

```diff
@@ -29,23 +29,40 @@
 using System.Reflection;
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
-
+using System.Threading;
 using ObjCRuntime;
 
 namespace Foundation {
 
 	public partial class NSDate {
-		const long NSDATE_TICKS = 631139040000000000;
+		const double NANOSECS_PER_MILLISEC = 1000000.0;
+
+		private static readonly NSCalendar calendar;
```

Comment:
This can be initialized inline (avoiding static constructor):
```csharp
static readonly NSCalendar calendar = new NSCalendar (NSCalendarType.Gregorian) { 
    TimeZone = NSTimeZone.FromName ("UTC"),
};
```

Reply:
Agreed.